src/test/librados/io_cxx.cc

```diff
@@ -218,6 +218,38 @@ TEST_F(LibRadosIoPP, SparseReadOpPP) {
   }
 }
 
+TEST_F(LibRadosIoPP, SparseReadExtentArrayOpPP) {
+  char buf[128], zbuf[128];
+  memset(buf, 0xcc, sizeof(buf));
+  memset(zbuf, 0, sizeof(zbuf));
+  bufferlist bl;
+  int i, len = 1024, skip = 5;
+  bl.append(buf, sizeof(buf));
+  for (i = 0; i < len; i++) {
+    if (!(i % skip)) {
+      ASSERT_EQ(0, ioctx.write("sparse-read", bl, bl.length(), i * sizeof(buf)));
+    }
+  }
+
+  bufferlist expect_bl;
+  for (i = 0; i < len; i++) {
+    if (!(i % skip)) {
+      expect_bl.append(buf, sizeof(buf));
+    } else if (i <= len - skip) {
+      expect_bl.append(zbuf, sizeof(zbuf));
+    }
+  }
+
+  std::map<uint64_t, uint64_t> extents;
+  bufferlist read_bl;
+  int rval = -1;
+  ObjectReadOperation op;
+  op.sparse_read(0, (len - (skip - 2)) * 128, &extents, &read_bl, &rval);
```

Comment:
`2` is a "magic number" -- if someone were to change the values of `len` and `skip`, this call wouldn't do the same thing as before and at least one of the asserts in `assert_eq_sparse()` would fire.

Since you are operating on 1021 blocks instead of 1024, why set `len` to 1024?  If you set `len` to 1021, you would be able to:
- make `else if (i <= len - skip)` just `else` in the loop above
- make `(len - (skip - 2)) * 128` just `len * 128` here

Reply:
Yeah, I have a better approach, which is always write the last segment to the file. Then we can easily set what the the block number to be.

Reply:
> I have a better approach, which is always write the last segment to the file.

I thought 1021, being an odd number, was somehow significant for the test.  If not, both approaches are essentially the same.